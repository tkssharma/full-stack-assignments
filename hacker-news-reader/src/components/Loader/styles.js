import styled, { keyframes } from 'styled-components';

const blink = keyframes`
  /**
  * At the start of the animation the dot
  * has an opacity of .2
  */
  0% {
    opacity: .2;
  }
  /**
  * At 20% the dot is fully visible and
  * then fades out slowly
  */
  20% {
    opacity: 1;
  }
  /**
  * Until it reaches an opacity of .2 and
  * the animation can start again
  */
  100% {
    opacity: .2;
  }
`;

export const Animation = styled.div`
  text-align: center;

  span {
    color: ${({ theme }) => theme.textSecondary};
    display: inline-block;
    margin-left: 4px;
    margin-right: 4px;
    font-size: 80px;
    line-height: 0.1;

    /**
    * Use the blink animation, which is defined above
    */
    animation-name: ${blink};
    /**
    * The total time of animation
    */
    animation-duration: 1s;
    /**
    * It will repeat itself forever
    */
    animation-iteration-count: infinite;
    /**
    * This makes sure that the starting style (opacity: .2)
    * of the animation is applied before the animation starts.
    * Otherwise we would see a short flash or would have
    * to set the default styling of the dots to the same
    * as the animation. Same applies for the ending styles.
    */
    animation-fill-mode: both;
  }

  span:nth-child(2) {
    animation-delay: 0.2s;
  }
  span:nth-child(3) {
    animation-delay: 0.4s;
  }
`;
