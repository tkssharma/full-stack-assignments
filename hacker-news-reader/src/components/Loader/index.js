import React from 'react';

import { Animation } from './styles';

const Loader = () => (
  <Animation>
    <span>.</span>
    <span>.</span>
    <span>.</span>
  </Animation>
);

export default Loader;
