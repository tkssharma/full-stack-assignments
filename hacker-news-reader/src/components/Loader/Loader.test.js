import React from 'react';
import { shallow } from 'enzyme';
import Loader from './index';

it('renders without crashing', () => {
  const component = shallow(<Loader />);
  expect(component).toMatchSnapshot();
});
