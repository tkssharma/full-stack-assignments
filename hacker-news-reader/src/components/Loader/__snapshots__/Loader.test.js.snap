// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <span>
          .
</span>,
        <span>
          .
</span>,
        <span>
          .
</span>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": ".",
        },
        "ref": null,
        "rendered": ".",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": ".",
        },
        "ref": null,
        "rendered": ".",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": ".",
        },
        "ref": null,
        "rendered": ".",
        "type": "span",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <span>
            .
</span>,
          <span>
            .
</span>,
          <span>
            .
</span>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": ".",
          },
          "ref": null,
          "rendered": ".",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": ".",
          },
          "ref": null,
          "rendered": ".",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": ".",
          },
          "ref": null,
          "rendered": ".",
          "type": "span",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
